jetty-http3/http3-qpack/src/main/java/org/eclipse/jetty/http3/qpack/QpackDecoder.java

```diff
@@ -0,0 +1,276 @@
+        _maxHeaderSize = maxHeaderSize;
+    }
+
+    QpackContext getQpackContext()
+    {
+        return _context;
+    }
+
+    public interface Handler
+    {
+        void onMetaData(long streamId, MetaData metadata);
+    }
+
+    public boolean decode(long streamId, ByteBuffer buffer, Handler handler) throws QpackException
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug("Decoding: streamId={}, buffer={}", streamId, BufferUtil.toDetailString(buffer));
+
+        // If the buffer is big, don't even think about decoding it
+        if (buffer.remaining() > _maxHeaderSize)
+            throw new QpackException.SessionException("431 Request Header Fields too large");
+
+        _integerDecoder.setPrefix(8);
+        int encodedInsertCount = _integerDecoder.decode(buffer);
+        if (encodedInsertCount < 0)
+            throw new QpackException.CompressionException("Could not parse Required Insert Count");
+
+        _integerDecoder.setPrefix(7);
+        boolean signBit = (buffer.get(buffer.position()) & 0x80) != 0;
+        int deltaBase = _integerDecoder.decode(buffer);
+        if (deltaBase < 0)
+            throw new QpackException.CompressionException("Could not parse Delta Base");
+
+        // Decode the Required Insert Count using the DynamicTable state.
+        DynamicTable dynamicTable = _context.getDynamicTable();
+        int insertCount = dynamicTable.getInsertCount();
+        int maxDynamicTableSize = dynamicTable.getCapacity();
+        int requiredInsertCount = decodeInsertCount(encodedInsertCount, insertCount, maxDynamicTableSize);
+
+        // Parse the buffer into an Encoded Field Section.
+        int base = signBit ? requiredInsertCount - deltaBase - 1 : requiredInsertCount + deltaBase;
+        EncodedFieldSection encodedFieldSection = new EncodedFieldSection(streamId, requiredInsertCount, base, buffer);
+
+        // Decode it straight away if we can, otherwise add it to the list of EncodedFieldSections.
+        if (requiredInsertCount <= insertCount)
+        {
+            _metaData = encodedFieldSection.decode(_context, _maxHeaderSize);
+            if (LOG.isDebugEnabled())
+                LOG.debug("Decoded: streamId={}, metadata={}", streamId, _metaData);
+
+            _instructions.add(new SectionAcknowledgmentInstruction(streamId));
+        }
+        else
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Deferred Decoding: streamId={}, encodedFieldSection={}", streamId, encodedFieldSection);
+            _encodedFieldSections.add(encodedFieldSection);
+        }
+
+        if (!_instructions.isEmpty())
+            _handler.onInstructions(_instructions);
+        _instructions.clear();
+
+        MetaData metaData = _metaData;
+        _metaData = null;
+        if (metaData != null)
+            handler.onMetaData(streamId, metaData);
+
+        return metaData != null;
+    }
+
+    void parseInstruction(ByteBuffer buffer) throws QpackException
```

Comment:
@lachlan-roberts is this public API?  is this the only way that recieved instructions make it into the instruction parser?